### ISSUE TITLE
WSL2 shut down without warning while I was using it, because Windows Store began to update it

### ISSUE THREAD

**Issue by astroboylrx** [microsoft/WSL#8169]
> **Is your feature request related to a problem? Please describe.**

I'm frustrated when WSL2 shut down itself without any warning while I'm still using it. And that's because Windows Store began to update it.

> **Describe the solution you'd like**

A warning of shutting down before updating it would be greatly appreciated (something like Office update, asking users save their work). I feel like WSL2, as a subsystem, should be treated more as a "system" instead of an app.

> **Describe alternatives you've considered**

It is convenient for the Store to auto-update some apps that are not running. However, for things that are running, especially WSL2, why does the Store has such a high priority of doing what it wants?

> **Additional context**

**Comment by benhillis** [comment]
Agreed - we've given this feedback to the store team and they've said it's something they ware working on. Thanks for the suggestion, will leave this open for tracking.

**Comment by Juan012345** [comment]
Store WSL

**Comment by dev0T** [comment]
This just happened with me. I use WSL as a working environment and had it close without any warning.

**Comment by astroboylrx** [comment]
This happened to me again today on a new machine where I forgot to turn off auto updates. I treat WSL as a working environment, not a toy... sigh

```
WSL version: 2.1.3.0
Kernel version: 5.15.146.1-2
WSLg version: 1.0.60
MSRDC version: 1.2.5105
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22635.3140
```

**Comment by astroboylrx** [comment]
Today it happened again. Microsoft Store updated WSL without my consent. And I already turned off auto updates in Store!

**Comment by slonopotamus** [comment]
Obviously, you didn't.

**Comment by astroboylrx** [comment]
> Obviously, you didn't.

How condescending. 
The only thing exposed to users is the toggle in Store's settings. I'd be glad to hear more if you know other measures.

**Comment by interj** [comment]
This issue is still present, System Components from Store ignore automatic update settings.

**Comment by jovere** [comment]
I'm having this problem as well. It's happened twice in as many weeks. If it would at least allow me to do a controlled shutdown *outside of my working hours*, I would let it. But instead it just kills all my applications. This is equivalent to just pulling the battery from a running laptop. How is this acceptable?

**Comment by muchcharles** [comment]
Just had this happen; why would this be the default?  Just randomly killing your sessions with no warning?

**Comment by bpsib** [comment]
So annoying I workaround this by using `tmux` when I want to use Windows Terminal so my sessions aren't lost and `conhost.exe` (legacy console) when I don't use a multiplexer.

EDIT: Somehow misread this so thought it was about Windows Terminal when it's about WSL2. My fault.

**Comment by MicobyteMichael** [comment]
This auto-update issue just shut down my WSL *while I was in the middle of running a backup*, and I have now lost several hours' productivity as a consequence. +1 to any effort that seeks to resolve this, because this is ridiculous and unacceptable.

**Comment by rubaiyat9** [comment]
I don't understand why this issue still exists more than 3 years later. If this is so hard to fix, at least allow us to turn off auto-update entirely until a proper solution is in place.

**Comment by 13BTurbo** [comment]
This is still an issue. Now I've lost unsaved work in VSCode because WSL did an update while it's being actively used.

**Comment by pratt-fds** [comment]
Ugh, this is a major problem for me - In my enterprise environment, I don't have the ability to shut off automatic updates and this issue is very disruptive to my workflow. How is this acceptable to Microsoft?

**Comment by ComBin** [comment]
Faced same problem just now. Really need fix it! Or some workaround.

**Comment by xmarkclx** [comment]
It's crazy they were working on it since 2022 and I still just encountered this today.
I tried to search how to turn off WSL2  store auto updates and there are no viable solutions.

**Comment by aggo15** [comment]
Just want to put [this discussion](https://github.com/microsoft/WSL/discussions/12675) under this issue. Obviously a lot of people are affected by this. Hopefully someone from Microsoft notice this issue and work on it asap.
